x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/ShardChangesIT.java

```diff
@@ -916,7 +916,7 @@ private void assertTotalNumberOfOptimizedIndexing(Index followerIndex, int numbe
                 for (String node : internalCluster().nodesInclude(followerIndex.getName())) {
                     IndicesService indicesService = internalCluster().getInstance(IndicesService.class, node);
                     IndexShard shard = indicesService.getShardOrNull(new ShardId(followerIndex, shardId));
-                    if (shard != null) {
+                    if (shard != null && shard.routingEntry().primary()) {
```

Comment:
We should not to restrict to primary shards because this assertion is executed inside assertBusy and we also need to verify the optimization of the replicas. However, let's improve this assertion later.